lms/djangoapps/mobile_api/tests/test_middleware.py

```diff
@@ -0,0 +1,147 @@
+        with mock.patch.object(caches['default'], 'get_many', wraps=caches['default'].get_many) as monkey:
+            processed_response = self.middleware.process_response(fake_request, request_response or HttpResponse())
+            self.assertEqual(0, monkey.call_count)
+        return request_response, processed_response
+
+    @ddt.data(
+        ("Mozilla/5.0 (Linux; Android 5.1; Nexus 5 Build/LMY47I; wv) AppleWebKit/537.36 (KHTML, like Gecko) "
+         "Version/4.0 Chrome/47.0.2526.100 Mobile Safari/537.36 edX/org.edx.mobile/2.0.0"),
+        ("Mozilla/5.0 (iPhone; CPU iPhone OS 9_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) "
+         "Mobile/13C75 edX/org.edx.mobile/2.2.1"),
+        ("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.116 "
+         "Safari/537.36"),
+    )
+    def test_non_mobile_app_requests(self, user_agent):
+        with self.assertNumQueries(0):
+            request_response, processed_response = self.process_middleware(user_agent)
+        self.assertIsNone(request_response)
+        self.assertEquals(200, processed_response.status_code)
+        self.assertNotIn(AppVersionUpgrade.LATEST_VERSION_HEADER, processed_response)
+        self.assertNotIn(AppVersionUpgrade.UPGRADE_DEADLINE_HEADER, processed_response)
+
+    @ddt.data(
+        "edX/org.edx.mobile (6.6.6; OS Version 9.2 (Build 13C75))",
+        "edX/org.edx.mobile (7.7.7; OS Version 9.2 (Build 13C75))",
+        "Dalvik/2.1.0 (Linux; U; Android 5.1; Nexus 5 Build/LMY47I) edX/org.edx.mobile/6.6.6",
+        "Dalvik/2.1.0 (Linux; U; Android 5.1; Nexus 5 Build/LMY47I) edX/org.edx.mobile/7.7.7",
+    )
+    def test_no_update(self, user_agent):
+        with self.assertNumQueries(2):
+            request_response, processed_response = self.process_middleware(user_agent)
+        self.assertIsNone(request_response)
+        self.assertEquals(200, processed_response.status_code)
+        self.assertNotIn(AppVersionUpgrade.LATEST_VERSION_HEADER, processed_response)
+        self.assertNotIn(AppVersionUpgrade.UPGRADE_DEADLINE_HEADER, processed_response)
+        with self.assertNumQueries(0):
+            self.process_middleware(user_agent)
+
+    @ddt.data(
+        "edX/org.edx.mobile (5.1.1; OS Version 9.2 (Build 13C75))",
+        "edX/org.edx.mobile (5.1.1.RC; OS Version 9.2 (Build 13C75))",
+        "Dalvik/2.1.0 (Linux; U; Android 5.1; Nexus 5 Build/LMY47I) edX/org.edx.mobile/5.1.1",
+        "Dalvik/2.1.0 (Linux; U; Android 5.1; Nexus 5 Build/LMY47I) edX/org.edx.mobile/5.1.1.RC",
+    )
+    def test_new_version_available(self, user_agent):
+        with self.assertNumQueries(2):
+            request_response, processed_response = self.process_middleware(user_agent)
+        self.assertIsNone(request_response)
+        self.assertEquals(200, processed_response.status_code)
+        self.assertEqual("6.6.6", processed_response[AppVersionUpgrade.LATEST_VERSION_HEADER])
```

Comment:
hmm.. are you aware of this number being the "number of the beast" and its biblical reference?